packages/storage/src/implementation/location.ts

```diff
@@ -76,23 +76,42 @@ export class Location {
       }
     }
     const gsPath = '(/(.*))?$';
-    const path = '(/([^?#]*).*)?$';
     const gsRegex = new RegExp('^gs://' + bucketDomain + gsPath, 'i');
     const gsIndices = { bucket: 1, path: 3 };
 
     function httpModify(loc: Location): void {
       loc.path_ = decodeURIComponent(loc.path);
     }
     const version = 'v[A-Za-z0-9_]+';
-    const hostRegex = DEFAULT_HOST.replace(/[.]/g, '\\.');
-    const httpRegex = new RegExp(
-      `^https?://${hostRegex}/${version}/b/${bucketDomain}/o${path}`,
+    const firebaseStorageHost = DEFAULT_HOST.replace(/[.]/g, '\\.');
+    const firebaseStoragePath = '(/([^?#]*).*)?$';
```

Comment:
What's special about `#`? Trying to understand the difference between `firebaseStoragePath` and `cloudStoragePath`.

Reply:
I added a test.